### DISCUSSION TITLE
Scanner scans slower and slower in version 1.9.23

### DISCUSSION THREAD

**Discussion by ToRa-51** [jareff-g/ALT-Scann8#108] [Q&A] [answered]
Hi Juan, 

Yesterday I tested ALT-Scann8 version 1.9.23 and scanning an R8 movie went well, but there was a strange phenomenon. The scanner starts scanning at a speed of about 130 frames/minute and at the end of the small reel it was only 57 frames/minute.
Incidentally, I had already noticed this phenomenon in previous versions, but have not done anything else with it because of other work on the scanner (new photo transistor and the Thingiverse Mod).
<img width="342" alt="Framesminute Frame number Movie time" src="https://github.com/jareff-g/ALT-Scann8/assets/26844127/141ab8be-cd48-4699-8a61-32a981f915d3">
Attached 2 files, ALT-Scann8.json and ALT-Scann8.20240218.log 
I'm not sure exactly how I could find these kinds of errors in the py file, but assume you can. 

Thanks in advance, 
Ton
[ALT-Scann8.20240218.log](https://github.com/jareff-g/ALT-Scann8/files/14334333/ALT-Scann8.20240218.log)
[ALT-Scann8.json](https://github.com/jareff-g/ALT-Scann8/files/14334335/ALT-Scann8.json)

P.S., I'm curious about the latest changes in the software (1.9.30) if they are also made public.

**Comment by jareff-g** [comment]
Hello Ton,
Thanks for your feedback.
I also experienced similar issues in the past, and delivered a fix in version 1.8.48, but since I have not done a lot of scans since then I don't know if the problem was really fixed.
What I can see in the log file is that the time required to save each frame is increasing as time passes, from 60 ms at the beginning to 180 ms at the end.
**One question**: If when the frames/minute fall to 57, you stop ALT-Scann8 (by stopping scan / exiting app), and restart it again, do you have the good frame rate (130 FPM) or it stays slow (57 FPM). With the problem as I detected it problem was solved by the restart. 
The latest changes I'm doing could have an influence in that problem as well, but it is not yet ready for general use.

Let me know
Rgds/Juan

**Reply by ToRa-51** [reply]
Hai Juan,

Yesterday I took the test again and performed it as you said. The result was: start at 4:08 PM (FPM 122) and after almost half an hour the speed had fallen back to 81 FPM. Then closed the app on the scanner and then turned off the scanner completely. After 15 minutes, restarted the scanner and the program and indeed the speed is back to the high FPM value. However, after about 15 minutes the speed had dropped again to 89 FPM. This test was done with a 7cm film roll and an R8 movie.

I saw on Github that there are now several branches, one of which is called Stable with version 1.9.12.
Can I use that to also do a test to see what the result is?

Regards, Ton.

**Reply by jareff-g** [reply]
Hello Ton,

In fact the test I wanted to do was to stop the ongoing scan, and immediately restart the app and start the scan again. But no problem, I'm pretty sure the original speed would have been reached in that scenario as well also. For the moment I'm working on other items, I'll try to look at this issue when i have time.
The 3 branches in Github:
- Develop: This is the one where I do my developments, even if I try to avoid putting non-working code there, it might happen. Also, changes are not documented so you might not be able to make it work without the details (this is the case with the current version in this branch)
- Stable: This branch will be updated from Master, once that the version in Master has bee n there for a while withthout any big issues being reported
- Master: This is the main branch, where I add the changes from develop once it looks they are working fine.
Feel free to try any of them. It can be useful to have a reference in cases like this

**Comment by jareff-g** [comment]
Yesterday I made a test (using version 1.10.0) regarding the decreasing speed issue, and in 30 minutes the speed went down from the initial 134 FPM to 120, and it seemed to stabilize there. At that point the RPi temperature was around 70℃, and even if at that temperature no throttling should happen, I put a fan in front of the scanner (box slightly opened) so the temperature fell to 50℃, with the speed staying at 120 FPM (so temperature is not a factor). I also took note of the overall memory usage in the RPi at the start/end of the test and there seems to be no memory leak (I'll check today memory used just by the ALT-Scann8 process).

Can you made a test in your scanner with version 1.10.0? I think you might find the the speed decrease is more in line with my tests, due to the latest changes I made.

**Reply by ToRa-51** [reply]
Hi Juan, 

Two days ago I was able to do a test with version 1.10.3 and it all looks promising. Partly by purchasing a new monitor (Samsung ViewFinity S6 LS24A600NAUXEN 24" Quad HD IPS monitor) everything is now a lot more readable than with that small screen I used for it.
Started scanning with Scan Speed 10, but because I thought it was going a bit too fast, I put it back to Scan Speed 7 a little later. The result is in the table below.
<img width="413" alt="Framesminute Frame number Movie time" src="https://github.com/jareff-g/ALT-Scann8/assets/26844127/4402397b-e0dd-4cf9-a99f-6fd190bc41ff">
Scanning the movie was done at once, so not stopped and restarted the app in the meantime.
It is strange that the number of FPM slowly drops from the start to 105 FPM and then rises again to 157 FPM and then drops again to 136 FPM until the end of the film.
If this also happens with a larger reel with film, I can be satisfied with it.
Attached the debug log file.
[ALT-Scann8.20240227.log](https://github.com/jareff-g/ALT-Scann8/files/14450745/ALT-Scann8.20240227.log)

**Reply by jareff-g** [reply]
Hello Ton,
That looks better to me. The variations you see are normal since (form what I see in the log file) you are using automatic exposure with 'Match wait' enabled. This causes ALT-Scann8 to wait for the automatic exposure to stabilize before capturing the frame. If you want to minimize that delay, and have a more homogeneous speed, you can uncheck the 'match wait' checkbox, or increase the match margin (however this will cause some visible effects in case of big exposure changes during scene transitions).
Regards,
Juan

**Reply by ToRa-51** [reply]
Hello Juan, 

Due to illness now finally be able to write a sequel to this discussion. 
On March 16, I was able to do a test with version 1.10.17 and the first thing I noticed was that the app set itself to a very nice format when it started up where everything is very easy to read. In any case, I am very happy with it. 
A small reel with R8 film scanned at once, it took 34 minutes to store 4749 frames, meaning the scanning speed averaged 140 FPM. Setting MatchWaitMargin increased to 80 according to your advice and I have increased the ScanSpeed a bit to 8. Scan result is to my satisfaction.

I will also do a test one of these days with the last uploaded 1.10.46 version.

Regards,
Ton.

**Reply by jareff-g** [reply, marked as answer]
Thanks for the feedback Ton.
ALT-Scann now tries to adjust the user interface size to the biggest possible size matching the desktop resolution, however you can change it by setting the 'main UI font size' in the new settings popup to other size you might prefer.
Also, scan speed is not displayed in frames per second, as it is a more standard measure.
Regards,
Juan

**Comment by jareff-g** [comment]
The slowdown issue was already solved, see [this issue](https://github.com/jareff-g/ALT-Scann8/issues/154) for details.
